src/__test__/utils/postErrorToSlack.test.js

```diff
@@ -0,0 +1,93 @@
+import postErrorToSlack from 'utils/postErrorToSlack';
+import fetchMock, { enableFetchMocks } from 'jest-fetch-mock';
+
+enableFetchMocks();
+
+const mockError = { stack: 'Mock error stack' };
+const mockInfo = { componentStack: 'Mock component stack' };
+const mockReduxDump = {
+  cellInfo: {
+    focus: {
+      store: 'cellSets',
+      key: 'louvain',
+    },
+    groupedTrack: 'louvain',
+    selectedTracks: [
+      'louvain',
+    ],
+  },
+  cellMeta: {
+    mitochondrialContent: {
+      loading: true,
+      error: false,
+      data: [],
+    },
+    doubletScores: {
+      loading: true,
+      error: false,
+      data: [],
+    },
+  },
+};
+
+jest.useFakeTimers('modern').setSystemTime(new Date('2020-01-01').getTime());
+
+delete window.location;
+window.location = {
+  href: 'http://localhost:3000/experiments/testae48e318dab9a1bd0bexperiment/data-exploration',
+};
+
+jest.mock('@aws-amplify/auth', () => ({
+  currentAuthenticatedUser: () => Promise.resolve({
+    attributes: {
+      name: 'John Doe',
+      email: 'fake@email.com',
+    },
+    username: '5152fake-eb52-474c-user-mocke8c8user',
+  }),
+}));
+
+describe('PostErrorToSlack', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    process.env.NODE_ENV = 'production';
```

Comment:
Having functions use environment variables makes it trickier to test them because they need global variables to be set and it's much harder to reason about the global state than just the local state. In this situations it's more convenient to have the state passed into the function ie:

`const postErrorToSlack = async (error, info, reduxDump, env) => {`
 
so then you can test them by simply:
```
await postErrorToSlack(mockError, mockInfo, mockReduxDump, 'staging');
await postErrorToSlack(mockError, mockInfo, mockReduxDump, 'production');
```

Reply:
Actually I'd rather have the calling function decide whether the to call the function or not, rather than passing in the environment variable to the function. This means,

    const ErrorBoundary = () => {
        ...

        if(process.env.NODE_ENV !== 'production') return;
        postErrorToSlack(error, errorInfo);

    }

Therefore, the only concern of `postErrorToSlack` is posting the error to Slack - it doesn't have to know to post or not.